assets/js/amp-customize-controls.js

```diff
@@ -0,0 +1,201 @@
+	};
+
+	/**
+	 * Create an non-AMP version of a URL.
+	 *
+	 * @param {string} url URL.
+	 * @return {string} non-AMPified URL.
+	 */
+	component.unampifyUrl = function unampifyUrl( url ) {
+		var urlParser = document.createElement( 'a' ),
+			regexEndpoint = new RegExp( '\\/' + component.data.query + '\\/?$' ),
+			params;
+
+		urlParser.href = url;
+		urlParser.pathname = urlParser.pathname.replace( regexEndpoint, '' );
+
+		if ( urlParser.search.length > 1 ) {
+			params = wp.customize.utils.parseQueryString( urlParser.search.substr( 1 ) );
+			delete params[ component.data.query ];
+			urlParser.search = $.param( params );
+		}
+
+		return urlParser.href;
+	};
+
+	/**
+	 * Create an AMP version of a URL.
+	 *
+	 * @param {string} url URL.
+	 * @return {string} AMPified URL.
+	 */
+	component.ampifyUrl = function ampifyUrl( url ) {
+		var urlParser = document.createElement( 'a' );
+		urlParser.href = component.unampifyUrl( url );
+		if ( urlParser.search.length ) {
+			urlParser.search += '&';
+		}
+		urlParser.search += component.data.query + '=1';
+		return urlParser.href;
+	};
+
+	/**
+	 * Hook up all AMP preview interactions once panel is ready.
+	 *
+	 * @param {wp.customize.Panel} panel The AMP panel.
+	 * @return {void}
+	 */
+	component.panelReady = function panelReady( panel ) {
+		/**
+		 * Make current URL AMPified if toggle is on.
+		 *
+		 * @param {string} url URL.
+		 * @return {string} AMPified URL.
+		 */
+		function setCurrentAmpUrl( url ) {
+			var enabled = api.state( 'ampEnabled' ).get();
+			if ( ! enabled && component.isAmpUrl( url ) ) {
+				return component.unampifyUrl( url );
+			} else if ( enabled && ! component.isAmpUrl( url ) ) {
+				return component.ampifyUrl( url );
+			}
+			return url;
+		}
+
+		/**
+		 * Swap to AMP version of URL in preview.
+		 *
+		 * @return {void}
+		 */
+		function updatePreviewUrl() {
+			api.previewer.previewUrl.set( setCurrentAmpUrl( api.previewer.previewUrl.get() ) );
+		}
+
+		// AMP panel triggers the input toggle for AMP preview.
+		panel.expanded.bind( function() {
+			if ( api.state( 'ampAvailable' ).get() ) {
+				api.state( 'ampEnabled' ).set( panel.expanded.get() );
+			}
+		} );
+
+		// Enable AMP toggle if available and mobile device selected.
+		api.previewedDevice.bind( function( device ) {
+			if ( 'mobile' === device && api.state( 'ampAvailable' ).get() ) {
+				api.state( 'ampEnabled' ).set( true );
+			}
+		} );
+
+		// Message coming from previewer.
+		api.previewer.bind( 'amp-status', function( data ) {
+			api.state( 'ampAvailable' ).set( data.available );
+		} );
+		function setInitialAmpEnabledState( data ) {
+			api.state( 'ampEnabled' ).set( data.enabled );
+			api.previewer.unbind( 'amp-status', setInitialAmpEnabledState );
+		}
+		api.previewer.bind( 'amp-status', setInitialAmpEnabledState );
+
+		// Persist the presence or lack of the amp=1 param when navigating in the preview,
+		// even if current page is not yet supported.
```

Comment:
Small detail, this doesn't follow WP JS code standards (see [here](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#comments)).